libmscore/durationtype.cpp

```diff
@@ -304,7 +304,11 @@ void TDuration::shiftType(int nSteps, bool stepDotted)
                   newValue = int(_val) + nSteps;
                   }
 
-            if ((newValue < int(DurationType::V_LONG)) || (newValue > int(DurationType::V_128TH)))
+            if ((newValue < int(DurationType::V_LONG)) || (newValue > int(DurationType::V_128TH)) ||
+                ((newValue == int(DurationType::V_128TH)) && (newDots >= 1)) ||
+                ((newValue == int(DurationType::V_64TH))  && (newDots >= 2)) ||
+                ((newValue == int(DurationType::V_32ND))  && (newDots >= 3)) ||
+                ((newValue == int(DurationType::V_16TH))  && (newDots >= 4)))
```

Comment:
There is no provision for 4 dots in MuseScore 2. In Chord, the NoteDot array has only 3 slots.